packages/material-ui/src/BottomNavigationAction/BottomNavigationAction.js

```diff
@@ -1,61 +1,113 @@
 import * as React from 'react';
 import PropTypes from 'prop-types';
 import clsx from 'clsx';
-import withStyles from '../styles/withStyles';
+import { deepmerge } from '@material-ui/utils';
+import { unstable_composeClasses as composeClasses } from '@material-ui/unstyled';
+import experimentalStyled from '../styles/experimentalStyled';
+import useThemeProps from '../styles/useThemeProps';
 import ButtonBase from '../ButtonBase';
 import unsupportedProp from '../utils/unsupportedProp';
+import bottomNavigationActionClasses, {
+  getBottomNavigationActionUtilityClass,
+} from './bottomNavigationActionClasses';
 
-export const styles = (theme) => ({
+const overridesResolver = (props, styles) => {
+  const { styleProps } = props;
+
+  return deepmerge(styles.root || {}, {
+    ...(!styleProps.showLabel && !styleProps.selected && styles.iconOnly),
+    [`& .${bottomNavigationActionClasses.wrapper}`]: styles.wrapper,
+    [`& .${bottomNavigationActionClasses.label}`]: styles.label,
+  });
+};
+
+const useUtilityClasses = (styleProps) => {
+  const { classes, showLabel, selected } = styleProps;
+
+  const slots = {
+    root: ['root', !showLabel && !selected && 'iconOnly', selected && 'selected'],
+    wrapper: ['wrapper'],
+    label: ['label', !showLabel && !selected && 'iconOnly', selected && 'selected'],
+  };
+
+  return composeClasses(slots, getBottomNavigationActionUtilityClass, classes);
+};
+
+const BottomNavigationActionRoot = experimentalStyled(
+  ButtonBase,
+  {},
+  {
+    name: 'MuiBottomNavigationAction',
+    slot: 'Root',
+    overridesResolver,
+  },
+)(({ theme, styleProps }) => ({
   /* Styles applied to the root element. */
-  root: {
-    transition: theme.transitions.create(['color', 'padding-top'], {
-      duration: theme.transitions.duration.short,
-    }),
-    padding: '6px 12px 8px',
-    minWidth: 80,
-    maxWidth: 168,
-    color: theme.palette.text.secondary,
-    flex: '1',
-    '&$iconOnly': {
+  transition: theme.transitions.create(['color', 'padding-top'], {
+    duration: theme.transitions.duration.short,
+  }),
+  padding: '6px 12px 8px',
+  minWidth: 80,
+  maxWidth: 168,
+  color: theme.palette.text.secondary,
+  flex: '1',
+  ...(!styleProps.showLabel &&
+    !styleProps.selected && {
       paddingTop: 16,
-    },
-    '&$selected': {
+    }),
+  ...(styleProps.selected && {
+    [`&.${bottomNavigationActionClasses.selected}`]: {
       paddingTop: 6,
       color: theme.palette.primary.main,
     },
+  }),
```

Comment:
```suggestion
  [`&.${bottomNavigationActionClasses.selected}`]: {
    paddingTop: 6,
    color: theme.palette.primary.main,
  },
```

No need for checking the `styleProps.selected` as the class is conditionally applied